api/python/provisioner/commands/destroy_vm.py

```diff
@@ -0,0 +1,157 @@
+)
+from ..utils import run_subprocess_cmd
+from ..salt import local_minion_id
+from ..pillar import (
+    PillarKey,
+    PillarResolver
+)
+from ..vendor import attr
+# TODO IMPROVE EOS-8473
+
+from .configure_setup import (
+    SetupType
+)
+from .deploy import (
+    build_deploy_run_args,
+    Deploy
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+deploy_states = dict(
+    prvsnr=[
+        "system.storage.glusterfs.teardown.volume_remove",
+        "system.storage.glusterfs.teardown.remove_dir",
+        "system.storage.glusterfs.teardown.services",
+        "system.storage.glusterfs.teardown.package_remove"
+    ]
+)
+
+run_args_type = build_deploy_run_args(deploy_states)
+
+
+@attr.s(auto_attribs=True)
+class DestroyVM(Deploy):
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+    _run_args_type = run_args_type
+
+    def _remove_prvsnr(self, nodes):
+        for node in nodes:
+            if local_minion_id() == node:
+                logger.info(f"remove cortx-prvsnr rpms on {node}")
+                run_subprocess_cmd(
+                    f"ssh {node} yum erase -y cortx-prvsnr")
+            else:
+                logger.info(
+                    f"remove cortx-prvsnr python36-cortx-prvsnr on {node}")
+                run_subprocess_cmd(
+                    f"ssh {node} yum erase -y"
+                    "cortx-prvsnr python36-cortx-prvsnr")
+
+    def _remove_passwordless_ssh(self, nodes):
+        for node in nodes:
+            logger.info(f"Remove passwordless configuration with {node}")
+            run_subprocess_cmd(
+                f"rm -rf /root/.ssh")
+
+    def _remove_salt(self, nodes):
+        salt_cache_dir = '/var/cache/salt'
+        salt_config = '/etc/salt'
+
+        for node in nodes:
+            logger.info(f"stopping salt-minion/salt-master on {node}")
+            run_subprocess_cmd(
+                f"ssh {node} systemctl stop salt-minion salt-master")
+
+            logger.info(f"remove salt rpms on {node}")
+            run_subprocess_cmd(
+                f"ssh {node} yum erase -y salt-minion salt-master salt")
+
+            logger.info(f"Removing salt from node {node}")
+            run_subprocess_cmd(f"ssh {node} rm -rf {salt_cache_dir}")
+            run_subprocess_cmd(f"ssh {node} rm -rf {salt_config}")
+
+    def _remove_dir(self, list_dir, nodes):
+        for node in nodes:
+            for di in list_dir:
+                logger.info(f"cleaning up {di} dir on {node}")
+                run_subprocess_cmd(f"ssh {node} rm -rf {di}")
+
+    def _run_states(self, states_group: str, run_args: run_args_type):
+        setup_type = run_args.setup_type
+        targets = run_args.targets
+        states = deploy_states[states_group]
+        stages = run_args.stages
+        primary = self._primary_id()
+        secondaries = f"not {primary}"
+        nodes = []
+
+        if states_group == 'prvsnr' and targets == ALL_MINIONS:
+            node_list = PillarKey('cluster/node_list')
```

Comment:
@ypise I guess it is going to be changed and `cluster/node_list` won't be available. Could you update on that?

Reply:
Yes, that's correct.
The srvnode-N keys need to be iterated as there would be no `cluster/node_list`

Reply:
comment is addressed